packages/common-common/src/rabbitmq/util.ts

```diff
@@ -0,0 +1,75 @@
+import fetch from "node-fetch";
+import {RascalQueues, RascalExchanges, RascalRoutingKeys} from './types';
+
+
+/*
+For more information on any of these routes and their behaviour/return values check:
+https://rawcdn.githack.com/rabbitmq/rabbitmq-server/v3.11.1/deps/rabbitmq_management/priv/www/api/index.html
+ */
+
+/**
+ * This function publishes a message containing the given data to the given exchange. In the event of a 400 Bad Request
+ * error ensure ALL nested objects in the data object are stringified BEFORE calling this function.
+ * @param rabbitMQUri The URI of the RabbitMQ Management API instance to connect to
+ * @param exchangeName The name of the exchange to publish the message to
+ * @param routingKey The routing key which defines what queue a message is routed to from the given exchange
+ * @param data The object to publish (WARNING: all nested objects must be individually json stringified)
+ * @returns {routed: boolean} If routed is false the message was not queued and if routed is true the message was queued
+ */
```

Comment:
I think these comments are helpful, but muddy the readability of the code, generally the code alone is readable and the functionality is implicit. These gotyas, could you include in the PR description instead of this long block of comments?

Reply:
I could but I had these because they provide better hints in IDEs when actually using the function. So their purpose is not necessarily meant to be when you're reviewing/reading the function definition itself but when you're calling it.